app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheet.kt

```diff
@@ -447,6 +463,12 @@ class BloodPressureEntrySheet : BottomSheetActivity(), BloodPressureEntryUi, Rem
     yearEditText.setTextAndCursor(fourDigitYear)
   }
 
+  override fun setDateOnInputFields(date: LocalDate) {
+    dayEditText.setTextAndCursor(dayDateFormatter.format(date))
+    monthEditText.setTextAndCursor(monthDateFormatter.format(date))
+    yearEditText.setTextAndCursor(fullYearDateFormatter.format(date))
```

Comment:
Instead of three different `DateTimeFormatters`, what do you think of injecting a single formatter that generates it in the `dd/MM/yyyy` format and then splitting it on the `/` character?

Reply:
Yeah, that can work. Although I feel like this is much more explicit than the splitting because then we have to get the item based on position.

Reply:
That's a good point. My intention was to reduce the number of injections we are doing purely for extracting date strings out of a date instance.

This could be a potential enhancement for the future, maybe we could consider creating something like a `DateInformationExtractor` class that abstracts this away so that it can reused between the two screens. What do you think?

Reply:
Yep, that sounds good. Tracking it [here](https://app.clubhouse.io/simpledotorg/story/3060/add-dateinformationextractor).